src/wp-includes/class-wp-scripts.php

```diff
@@ -871,8 +883,24 @@ private function get_dependents( $handle ) {
 	 */
 	private function get_intended_strategy( $handle ) {
 		$script_args = $this->get_data( $handle, 'script_args' );
+		$strategy    = isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
```

Comment:
What happens here is someone passes an empty string or `false` as a strategy? Do we ever hit this codepath or do we bail beforehand?

Reply:
@joemcgill passing an empty or otherwise _illegal_ value will have no ill bearing on the process. If no strategy is passed, it will still proceed to the next part of logic, which calls `$this->is_valid_strategy( $strategy )`. At this point, it will fail and return `false`, and the default blocking strategy will be used as part of the `get_eligible_loading_strategy()` method call.

Arguably, we are probably better off with a `! empty( $script_args['strategy'] )` as opposed to the current `isset( $script_args['strategy'] )` check, as an empty string would still default to `true` for an `isset()` check, but either one is not changing the final outcome 2 lines down from the check.

Reply:
The only side effect of using `! empty()` would be if `'strategy'` as a key was never set, but that is pretty much impossible in our logic given that we normalize the args. Either way though, what we have now is logical so I'm going to proceed with this merge, so that we have a _final_ codebase in the base PR to do one final test against.